go/vt/vtexplain/testdata/multi-output/selectsharded-output.txt

```diff
@@ -91,10 +91,10 @@ select name, count(*) from user group by name /* scatter aggregate */
 ----------------------------------------------------------------------
 select 1, "hello", 3.14, null from user limit 10 /* select constant sql values */
 
-1 ks_sharded/-40: select 1, 'hello', 3.14, null from `user` limit 10 /* INT64 */ /* select constant sql values */
-1 ks_sharded/40-80: select 1, 'hello', 3.14, null from `user` limit 10 /* INT64 */ /* select constant sql values */
-1 ks_sharded/80-c0: select 1, 'hello', 3.14, null from `user` limit 10 /* INT64 */ /* select constant sql values */
-1 ks_sharded/c0-: select 1, 'hello', 3.14, null from `user` limit 10 /* INT64 */ /* select constant sql values */
+1 ks_sharded/-40: select 1, 'hello', cast(3.14 as DECIMAL(3, 2)), null from `user` limit 10 /* INT64 */ /* select constant sql values */
```

Comment:
So the original idea I had here, was that that we don't need this case for a decimal literal where the scale != 0. `3.14` is already a decimal literal with the same type here, so we can avoid this one if we want to.